pkg/auth/api/types.go

```diff
@@ -7,10 +7,10 @@ import (
 // UserIdentityInfo contains information about an identity.  Identities are distinct from users.  An authentication server of
 // some kind (like oauth for example) describes an identity.  Our system controls the users mapped to this identity.
 type UserIdentityInfo interface {
```

Comment:
This isn't used how you'd expect.  `GetProviderName()` is never called.  That suggests that this is the wrong abstraction